src/components/EditNodeCptModal/component.js

```diff
@@ -0,0 +1,86 @@
+import React, { Fragment, useState, useCallback } from 'react';
+
+import Button from 'components/Button';
+import Modal from 'components/Modal';
+import PropTypes from 'prop-types';
+import { nodePropTypes } from 'models';
+import { isEnterKey } from 'utils/event';
+import { isNodeCptValid } from 'validations/node';
+import NodeCptParentStatesTable from 'components/NodeCptParentStatesTable';
+import NodeCptEditTable from 'components/NodeCptEditTable';
+import styles from './styles.scss';
+
+const handleCptKeyUpCreator = handleSave => (e) => {
+  if (isEnterKey(e)) {
+    handleSave();
+  }
+};
+
+const handleSaveCreator = (cpt, onSave, onAlert) => {
+  if (isNodeCptValid(cpt)) {
+    onSave(cpt);
+  } else {
+    onAlert('A soma das probabilidades para cada uma das linhas deve ser igual a 1');
+  }
+};
+
+const EditNodeCptModal = ({
+  node,
+  hasParents,
+  onSave,
+  onCancel,
+  onAlert,
+}) => {
+  const { id, states } = node;
+  const [cpt, setCpt] = useState(node.cpt);
+  const handleSave = useCallback(
+    () => handleSaveCreator(cpt, onSave, onAlert),
```

Comment:
I would also curry this `handleSaveCreator` function, as you are doing for `handleCptKeyUpCreator`, so you don't have to pass an anonymous function here.

```suggestion
    useCallback(handleSaveCreator(cpt, onSave, onAlert), [cpt])
```

```js
	const handleSaveCreator = (cpt, onSave, onAlert) => () => {}
```